File: truyenqq/src/genre.lua
local doc = html:parse("<a href=\"action-26.html\">Action</a><a href=\"adult-68.html\">Adult</a><a href=\"adventure-27.html\">Adventure</a><a href=\"anime-62.html\">Anime</a><a href=\"chuyen-sinh-91.html\">Chuyển Sinh</a><a href=\"co-dai-90.html\">Cổ Đại</a><a href=\"comedy-28.html\">Comedy</a><a href=\"comic-60.html\">Comic</a><a href=\"demons-99.html\">Demons</a><a href=\"detective-100.html\">Detective</a><a href=\"doujinshi-96.html\">Doujinshi</a><a href=\"drama-29.html\">Drama</a><a href=\"dam-my-93.html\">Đam Mỹ</a><a href=\"ecchi-50.html\">Ecchi</a><a href=\"fantasy-30.html\">Fantasy</a><a href=\"gender-bender-45.html\">Gender Bender</a><a href=\"harem-47.html\">Harem</a><a href=\"historical-51.html\">Historical</a><a href=\"horror-44.html\">Horror</a><a href=\"huyen-huyen-468.html\">Huyền Huyễn</a><a href=\"isekai-85.html\">Isekai</a><a href=\"josei-54.html\">Josei</a><a href=\"mafia-69.html\">Mafia</a><a href=\"magic-58.html\">Magic</a><a href=\"manhua-35.html\">Manhua</a><a href=\"manhwa-49.html\">Manhwa</a><a href=\"martial-arts-41.html\">Martial Arts</a><a href=\"mature-48.html\">Mature</a><a href=\"military-101.html\">Military</a><a href=\"mystery-39.html\">Mystery</a><a href=\"ngon-tinh-87.html\">Ngôn Tình</a><a href=\"one-shot-95.html\">One shot</a><a href=\"psychological-40.html\">Psychological</a><a href=\"romance-36.html\">Romance</a><a href=\"school-life-37.html\">School Life</a><a href=\"sci-fi-43.html\">Sci-fi</a><a href=\"seinen-42.html\">Seinen</a><a href=\"shoujo-38.html\">Shoujo</a><a href=\"shoujo-ai-98.html\">Shoujo Ai</a><a href=\"shounen-31.html\">Shounen</a><a href=\"shounen-ai-86.html\">Shounen Ai</a><a href=\"slice-of-life-46.html\">Slice of life</a><a href=\"smut-97.html\">Smut</a><a href=\"sports-57.html\">Sports</a><a href=\"supernatural-32.html\">Supernatural</a><a href=\"tragedy-52.html\">Tragedy</a><a href=\"trong-sinh-82.html\">Trọng Sinh</a><a href=\"truyen-mau-92.html\">Truyện Màu</a><a href=\"webtoon-55.html\">Webtoon</a><a href=\"xuyen-khong-88.html\">Xuyên Không</a><a href=\"yaoi-70.html\">Yaoi</a><a href=\"yuri-75.html\">Yuri</a>")
local genre = {}
if doc ~= nil then
    local el = doc:select("a")
    for i = 0, el:size() - 1 do
        local e = el:get(i)
        local link = {
            ["title"] = e:text(),
            ["input"] = "http://truyenqq.com/the-loai/" .. e:attr("href"),
            ["script"] = "gen.lua"
        }
        table.insert(genre, link)
    end
    return response:success(genre)
end

return nil

File: blogtruyen/src/genre.js
function execute() {
    var genre = [];
    var htm = "<li id=\"54\">16+</li><li id=\"45\">18+</li><li id=\"1\">Action</li><li id=\"2\">Adult</li><li id=\"3\">Adventure</li><li id=\"4\">Anime</li><li id=\"5\">Comedy</li><li id=\"6\">Comic</li><li id=\"7\">Doujinshi</li><li id=\"49\">Drama</li><li id=\"48\">Ecchi</li><li id=\"60\">Event BT</li><li id=\"50\">Fantasy</li><li id=\"64\">Full màu</li><li id=\"61\">Game</li><li id=\"51\">Gender Bender</li><li id=\"12\">Harem</li><li id=\"13\">Historical</li><li id=\"14\">Horror</li><li id=\"63\">Isekai/Dị giới/Trọng sinh</li><li id=\"15\">Josei</li><li id=\"16\">Live action</li><li id=\"46\">Magic</li><li id=\"55\">manga</li><li id=\"17\">Manhua</li><li id=\"18\">Manhwa</li><li id=\"19\">Martial Arts</li><li id=\"20\">Mature</li><li id=\"21\">Mecha</li><li id=\"22\">Mystery</li><li id=\"56\">Nấu Ăn</li><li id=\"65\">Ngôn Tình</li><li id=\"62\">NTR</li><li id=\"23\">One shot</li><li id=\"24\">Psychological</li><li id=\"25\">Romance</li><li id=\"26\">School Life</li><li id=\"27\">Sci-fi</li><li id=\"28\">Seinen</li><li id=\"29\">Shoujo</li><li id=\"30\">Shoujo Ai</li><li id=\"31\">Shounen</li><li id=\"32\">Shounen Ai</li><li id=\"33\">Slice of life</li><li id=\"34\">Smut</li><li id=\"35\">Soft Yaoi</li><li id=\"36\">Soft Yuri</li><li id=\"37\">Sports</li><li id=\"38\">Supernatural</li><li id=\"39\">Tạp chí truyện tranh</li><li id=\"40\">Tragedy</li><li id=\"58\">Trap (Crossdressing)</li><li id=\"57\">Trinh Thám</li><li id=\"41\">Truyện scan</li><li id=\"66\">Tu chân - tu tiên</li><li id=\"53\">Video Clip</li><li id=\"42\">VnComic</li><li id=\"52\">Webtoon</li><li id=\"59\">Yuri</li>";

    var el = Html.parse(htm).select("li");
    for (var i = 0; i < el.size(); i++) {
        var e = el.get(i);
        genre.push({
            title: e.text(),
            input: e.attr("id"),
            script: "gen.js"
        });
    }

    return Response.success(genre);
}
